pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -337,49 +142,835 @@
+            flagPrefix = "lib";
+          };
+          v4l2 = {
+            gate = stdenv.isLinux;
+            packages = { inherit libv4l; };
+            flags = [ "libv4l2" "v4l2-m2m" ];
+            description = "Video 4 Linux";
+          };
+          vaapi = {
+            gate = with stdenv; isLinux || isFreeBSD;
+            packages = {
+              libva = if ffmpegVariant == headless then libva-minimal else libva;
+            };
+            description = "Vaapi hardware acceleration";
+          };
+          vdpau = {
+            gate = !stdenv.hostPlatform.isMinGW;
+            variant = small;
+            packages = { inherit libvdpau; };
+            description = "Vdpau hardware acceleration";
+          };
+          videotoolbox = {
+            gate = stdenv.isDarwin;
+            packages = { inherit VideoToolbox; };
+          };
+          vidstab = {
+            gate = enable.gpl;
+            variant = full;
+            packages = { inherit vid-stab; };
+            flagPrefix = "lib";
+            description = "Video stabilization";
+          };
+          vmaf = {
+            gate = !stdenv.isAarch64;
+            variant = full;
+            version = "5";
+            packages = { inherit libvmaf; };
+            flagPrefix = "lib";
+            description = "Netflix's VMAF (Video Multi-Method Assessment Fusion)";
+          };
+          vo-amrwbenc = {
+            gate = enable.gplv3;
+            variant = full;
+            packages = { inherit vo-amrwbenc; };
+            flagPrefix = "lib";
+          };
+          vorbis = {
+            packages = { inherit libvorbis; };
+            flagPrefix = "lib";
+            description = "Vorbis de/encoding, native encoder exists"; # TODO shouldn't we be using it then?
+          };
+          vpl = {
+            # It is currently unclear whether this breaks support for old GPUs. See
+            # https://github.com/NixOS/nixpkgs/issues/303074
+            gate = false;
+            packages = { inherit libvpl; };
+            flagPrefix = "lib";
+            description = "Hardware acceleration via intel libvpl (incompatible with {option}`mfx`)";
+          };
+          vpx = {
+            gate = stdenv.buildPlatform == stdenv.hostPlatform;
+            packages = { inherit libvpx; };
+            flagPrefix = "lib";
+            description = "VP8 & VP9 de/encoding";
+          };
+          vulkan = {
+            gate = !stdenv.isDarwin;
+            variant = small;
+            packages = { inherit vulkan-headers vulkan-loader; }; };
+          webp = {
+            variant = full;
+            packages = { inherit libwebp; };
+            flagPrefix = "lib";
+          };
+          x264 = {
+            gate = enable.gpl;
+            packages = { inherit x264; };
+            flagPrefix = "lib";
+          };
+          x265 = {
+            gate = enable.gpl;
+            packages = { inherit x265; };
+            flagPrefix = "lib";
```

Comment:
Somehow `flagPrefix` doesn't seem to work. With this shell.nix
```
{ pkgs ? import ../nixpkgs {} }:
pkgs.mkShell {
  packages = [
    (
      pkgs.ffmpeg.override {
        configuration = {
          features.gpl.enable = true;
          features.x265.enable = true;
        };
      }
    )
  ];
}
```
I get
```
       > configure flags: --disable-static --prefix=/nix/store/la3im0f739w6faba4k2h35mlambdciii-ffmpeg-6.1.1 --target_os=linux --arch=x86_64 --pkg-config=pkg-config --datadir=/nix/store/r6z9z0qrjs7blsphfc2z1licwmip7sac-ffmpeg-6.1.1-data/share/ffmpeg --bindir=/nix/store/kwxif6j3p3138g12dvvfyjl8mnvxpxms-ffmpeg-6.1.1-bin/bin --libdir=/nix/store/dx9j1hi5pxqag8l7ns4lwpywfb7h9wxa-ffmpeg-6.1.1-lib/lib --incdir=/nix/store/zmn11vw7kzh205yl3m09wsnjb72dpq95-ffmpeg-6.1.1-dev/include --mandir=/nix/store/v9hyndmvzpirv0ayv77ak9gg8m88ld56-ffmpeg-6.1.1-man/share/man --docdir=/nix/store/mivsjfqgnj1difymzwnyvzzivsz20qs6-ffmpeg-6.1.1-doc/share/doc/ffmpeg --enable-alsa --disable-libaom --disable-appkit --disable-libaribcaption --enable-libass --disable-audiotoolbox --enable-avcodec --enable-avdevice --enable-avfilter --enable-avformat --disable-avfoundation --disable-avisynth --enable-avutil --disable-libbluray --disable-libbs2b --enable-bzlib --disable-libcaca --disable-libcelt --disable-chromaprint --disable-libcodec2 --disable-coreimage --disable-cuda --disable-cuda-llvm --enable-cuvid --enable-libdav1d --disable-libdc1394 --disable-debug --enable-doc --enable-libdrm --disable-extra-warnings --disable-libfdk-aac --enable-ffmpeg --enable-ffplay --enable-ffprobe --disable-libflite --enable-fontconfig --enable-libfontconfig --enable-libfreetype --disable-frei0r --disable-libfribidi --disable-libgme --enable-gnutls --enable-gpl --disable-gray --disable-libgsm --enable-hardcoded-tables --enable-libharfbuzz --enable-htmlpages --enable-iconv --disable-libjack --disable-libjxl --disable-ladspa --enable-lzma --enable-manpages --disable-libmfx --disable-libmodplug --enable-libmp3lame --disable-libmysofa --enable-network --enable-ffnvcodec --enable-nvdec --enable-nvenc --disable-openal --disable-opencl --disable-libopencore-amrnb --disable-libopencore-amrwb --disable-opengl --disable-libopenh264 --disable-libopenjpeg --disable-libopenmpt --enable-optimizations --enable-libopus --disable-os2threads --enable-pic --enable-pixelutils --disable-libplacebo --enable-podpages --enable-postproc --enable-pthreads --enable-libpulse --disable-librav1e --disable-librtmp --enable-runtime-cpudetect --enable-safe-bitstream-reader --disable-libsmbclient --enable-sdl2 --disable-libshaderc --enable-shared --disable-small --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --disable-static --disable-stripping --disable-librsvg --enable-libsvtav1 --enable-swresample --enable-swscale --enable-swscale-alpha --disable-libtensorflow --enable-libtheora --disable-thumb --enable-txtpages --disable-nonfree --enable-libv4l2 --enable-v4l2-m2m --enable-vaapi --enable-vdpau --enable-version3 --disable-videotoolbox --disable-libvidstab --disable-libvmaf --disable-libvo-amrwbenc --enable-libvorbis --disable-libvpl --enable-libvpx --enable-vulkan --disable-w32threads --disable-libwebp --enable-libx264 --enable-x265 --disable-libxavs --disable-libxcb --disable-libxcb-shape --disable-libxcb-shm --disable-libxcb-xfixes --disable-xlib --disable-libxml2 --enable-libxvid --enable-libzimg --enable-zlib --disable-libzmq
       > Unknown option "--enable-x265".
       > See ./configure --help for available options.
```

Reply:
Overriding one value of an ffmpegFlag appears to override the entire ffmpegFlag. Back to the drawing board...

Perhaps we do need https://github.com/NixOS/nixpkgs/pull/296979 for this.

Reply:
The fix wasn't as bad as I imagined.